content/2016.06.ab-testing-with-kinto.rst

```diff
@@ -0,0 +1,199 @@
+still a really good idea to test a feature.
+
+.. image:: {filename}/images/kinto-logo.png
+    :alt: The Kinto Logo
+
+
+In an other hand, I am a big fan of the philosophy behind Kinto_ which
+as you may know is a database for the web.
+
+In this article, I will take you through a little experiment I did
+lately to extend the `work that have been done`_ to activate a feature
+for some users in Firefox for Android.
+
+.. _`we can do better`: http://stevehanov.ca/blog/index.php?id=132
+.. _`certain circumstances`: https://vwo.com/blog/multi-armed-bandit-algorithm/
+.. _`work that have been done`: https://github.com/mozilla-services/switchboard-experiments/
+
+
+How does A/B testing works?
+===========================
+
+You just created a really nice feature and you want to test multiple
+UX solution in order to decide which one is most usable for your users.
+
+For instance, what is the best wording for the french banner?
+
+You can use A/B testing to test differents variations on your user
+base and select the best option.
+
+There is two technical parts on this:
+
+1. You need to display different version of your button to your users,
+   probably with the same code.
+2. You need to be able to get the conversion rate for each solution to
+   be able to compare them.
+
+
+How to use Kinto to know in which test your user is?
+====================================================
+
+The benefit of using Kinto_ is that you will be able to use the
+`kinto-admin`_ to configure your experiments.
+
+.. image:: {filename}/images/kinto-admin-screenshot.png
+    :alt: Adding an experiment with the kinto-admin.
+
+An experiment has got a name and a description as well as some
+matching rules and a buckets range.
+
+For the web, in my proof of concept, matching rules can be defined as
+regexp on the user languages, country or user-agent.
+
+Because it is easier to think in pourcentages, we split users in 100
+buckets with a bucketID from 0 to 99.
+
+In our experiment we can decide to include a range of buckets using
+the min and max buckets attributes.
+
+Using `fingerprint.js`_, it is possible to calculate the user bucket
+and see if they are within the experiment buckets range.
+
+.. code-block:: javascript
+
+     const deviceID = parseInt(fingerprint.deviceID, 16);
+	 const userBucket = deviceID % 100;
+
+An experiment is then stored in kinto like that (I added matching rules for the demo but they are all optional.)
+
+.. code-block:: json
+
+    {
+        "name": "french-banner",
+        "description": "Try the new french banner",
+        "match": {
+            "country": "FR",
+			"lang": "fr"
+        },
+        "buckets": {
+            "min": "0",
+            "max": "50"
+        },
+		"values": {
+            "url": "https://firefoxos.mozfr.org/dotclear/public/Firefox_OS/.Gerez_votre_vie_privee_intelligemment_Mozilla_m.png"
+		}
+    }
+
+
+So for a given experiment we have two crutial information that we can get from that:
```

Comment:
information is a mass noun, so "two pieces of information". "crucial" can go before pieces, or before information (I don't think it makes a major semantic difference). "get from that" doesn't sound right to me. How about rephrasing the whole sentence as "So a given experiment has two crucial pieces of information"?